tty/pc-tty/ttypc_vtf.c

```diff
@@ -634,212 +656,340 @@ void _ttypc_vtf_sgr(ttypc_virt_t *virt)
 		case 46:
 		case 47:
 			if (ttypc->color) {
-				colortouched = 1;
-				setcolor |= ((bgansitopc[(virt->parms[i - 1] - 40) & 7]) << 8);
+				cc = 1;
+				attr = (bgansitopc[(vt->parms[i - 1] - 40) & 7] << 8) | (attr & 0x0f00);
 			}
 			break;
 		}
-	} while (i <= virt->parmi);
+	} while (i <= vt->parmi);
 	
-	if (ttypc->color) {
-		if (colortouched)
-			virt->attr = setcolor;
-		else
-			virt->attr = ((sgr_tab_color[virt->vtsgr]) << 8);
-	}
-	else
-		virt->attr = ((sgr_tab_mono[virt->vtsgr]) << 8);
+	vt->attr = (ttypc->color) ? ((cc) ? attr : csgr[vt->sgr] << 8) : msgr[vt->sgr] << 8;
 }
 
 
-/* device status reports */
-void _ttypc_vtf_dsr(ttypc_virt_t *virt)
+/* Device status reports */
+void _ttypc_vtf_dsr(ttypc_vt_t *vt)
 {
-#if 0
-	static uint8_t *answr = (uint8_t *)"\033[0n";
-	static uint8_t *panswr = (uint8_t *)"\033[?13n";      /* Printer Unattached */
-	static uint8_t *udkanswr = (uint8_t *)"\033[?21n";    /* UDK Locked */
-	static uint8_t *langanswr = (uint8_t *)"\033[?27;1n"; /* North American*/
-#endif
-#if 0
-	static uint8_t buffer[16];
-
+	static char *stat  = "\033[0n";     /* Status */
+	static char *print = "\033[?13n";   /* Printer Unattached */
+	static char *udk   = "\033[?21n";   /* UDK Locked */
+	static char *lang  = "\033[?27;1n"; /* North American */
+	char buff[16];
 	int i = 0;
 
-	switch (virt->parms[0]) {
-	
-	/* return status */
+	switch (vt->parms[0]) {
+	/* Status */
 	case 5:
-		/* respond(answr); */
+		write(0, stat, sizeof(stat));
 		break;
 
-	/* return cursor position */
+	/* Cursor position */
 	case 6:
-		buffer[i++] = 0x1b;
-		buffer[i++] = '[';
+		buff[i++] = 0x1b;
+		buff[i++] = '[';
 		
-		if (virt->row + 1 > 10)
-			buffer[i++] = (virt->row + 1) / 10 + '0';
-		buffer[i++] = (virt->row + 1) % 10 + '0';
-		buffer[i++] = ';';
-
-		if (virt->col + 1 > 10)
-			buffer[i++] = (virt->col + 1) / 10 + '0';
-		buffer[i++] = (virt->col + 1) % 10 + '0';
-		buffer[i++] = 'R';
-		buffer[i++] = '\0';
-
-		/* respond(buffer) */;
+		if (vt->crow + 1 > 10)
+			buff[i++] = '0' + (vt->crow + 1) / 10;
+		buff[i++] = '0' + (vt->crow + 1) % 10;
+		buff[i++] = ';';
+
+		if (vt->ccol + 1 > 10)
+			buff[i++] = '0' + (vt->ccol + 1) / 10;
+		buff[i++] = '0' + (vt->ccol + 1) % 10;
+		buff[i++] = 'R';
+		buff[i++] = '\0';
+
+		write(0, buff, i);
 		break;
 
-	/* return printer status */
+	/* Printer status */
 	case 15:
-		/* respond(panswr); */
+		write(0, print, sizeof(print));
```

Comment:
`sizeof(print) = 4` (`sizeof(char *)`). It should be `static const char print[]`

Reply:
Fixed.